it/test/uk/gov/hmrc/alcoholdutycalculator/RatesIntegrationSpec.scala

```diff
@@ -55,4 +66,188 @@ class RatesIntegrationSpec extends ISpecBase {
+          "show the correct rate" in {
+            verifyRate(cy, taxTypes, count, rate)
+          }
+        }
+      }
+
+      "For band E" - {
+        val rate: Option[Double]  = Some(28.50d)
+        val taxTypes: Seq[String] = Seq("331", "333", "334", "335")
+        val count                 = 4
+
+        "service rates endpoint must" - {
+          "show the correct rate" in {
+            verifyRate(cy, taxTypes, count, rate)
+          }
+        }
+      }
+
+      "For band F" - {
+        val rate: Option[Double]  = Some(31.64d)
+        val taxTypes: Seq[String] = Seq("341", "343", "344", "345")
+        val count                 = 4
+
+        "service rates endpoint must" - {
+          "show the correct rate" in {
+            verifyRate(cy, taxTypes, count, rate)
+          }
+        }
+      }
+
+      "For band G" - {
+        val rate: Option[Double]  = Some(8.42d)
+        val taxTypes: Seq[String] = Seq("351", "352", "353", "354", "355")
+        val count                 = 5
+
+        "service rates endpoint must" - {
+          "show the correct rate" in {
+            verifyRate(cy, taxTypes, count, rate)
+          }
+        }
+      }
+
+      "For band H" - {
+        val rate: Option[Double]  = Some(19.08d)
+        val taxTypes: Seq[String] = Seq("356")
+        val count                 = 1
+
+        "service rates endpoint must" - {
+          "show the correct rate" in {
+            verifyRate(cy, taxTypes, count, rate)
+          }
+        }
+      }
+
+      "For band I" - {
+        val rate: Option[Double]  = Some(8.78d)
+        val taxTypes: Seq[String] = Seq("357")
+        val count                 = 1
+
+        "service rates endpoint must" - {
+          "show the correct rate" in {
+            verifyRate(cy, taxTypes, count, rate)
+          }
+        }
+      }
+
+      "For band J" - {
+        val rate: Option[Double]  = Some(19.08d)
+        val taxTypes: Seq[String] = Seq("358", "359", "360")
+        val count                 = 3
+
+        "service rates endpoint must" - {
+          "show the correct rate" in {
+            verifyRate(cy, taxTypes, count, rate)
+          }
+        }
+      }
+
+      "For band K" - {
+        val taxTypes: Seq[String] = Seq("371", "372", "373", "374", "375", "376", "377", "378", "379", "380")
+        val count                 = 10
+
+        "service rates endpoint must" - {
+          "show the correct rate" in {
+            verifyRate(cy, taxTypes, count, None)
+          }
+        }
+      }
+
+    }
+
+    // TODO when we hit the next tax year current cy tests become cy-1 and cy can be updated with new values (up to cy-5)
+    s"For cy-1" - {
+      "For band A" - {}
+      "For band B" - {}
+      "For band C" - {}
+      // etc...
+    }
+  }
+
+  private def verifyRate(taxYear: Int, taxTypes: Seq[String], count: Int, rate: Option[Double]): Unit = {
+    val rateBandList = getRatesFromJson(taxYear)
```

Comment:
Can we call this once per each Rate period just after `lazy val cy = 2023`?

Reply:
The call needs to be stubbed within the test at this point. Will look at a potential refactor.